src/system_reader.rs

```diff
@@ -699,6 +826,241 @@ mod tests {
+        );
+        // We step over the LST...
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Struct));
+        // ...but expect all of the symbols we encounter after it to be in the symbol table,
+        // indicating that the SystemReader processed the LST even though we skipped it with `next()`
+        assert_eq!(reader.next()?, Value(IonType::Symbol));
+        assert_eq!(reader.read_symbol()?, "foo");
+        Ok(())
+    }
+
+    #[test]
+    fn shared_symbol_table_import_with_max_id() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        map_catalog.put_table(SharedSymbolTable::new(
+            "shared_table".to_string(),
+            1,
+            vec![Some("foo".to_string())],
+        )?);
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+            $ion_symbol_table::{
+                imports: [ { name:"shared_table", version: 1, max_id: 1 }],
+            }
+            $10 // "foo"
+          "#,
+            Box::new(map_catalog),
+        );
+        // We step over the LST...
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Struct));
+        // ...but expect all of the symbols we encounter after it to be in the symbol table,
+        // indicating that the SystemReader processed the LST even though we skipped it with `next()`
+        assert_eq!(reader.next()?, Value(IonType::Symbol));
+        assert_eq!(reader.read_symbol()?, "foo");
+        Ok(())
+    }
+
+    #[test]
+    fn shared_symbol_table_step_in() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        map_catalog.put_table(SharedSymbolTable::new(
+            "shared_table".to_string(),
+            1,
+            vec![Some("foo".to_string())],
+        )?);
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+            $ion_symbol_table::{
+                imports: [ { name:"shared_table", version: 1 }],
+            }
+            $10 // "foo"
+          "#,
+            Box::new(map_catalog),
+        );
+        // We process the LST by stepping into it...
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Struct));
+        reader.step_in()?;
+        // Process the LST imports list
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::List));
+        reader.step_in()?;
+        // Process a shared symbol table import
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Struct));
+        reader.step_in()?;
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::String));
+        assert_eq!(reader.read_string()?, "shared_table");
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Int));
+        assert_eq!(reader.read_i64()?, 1);
+        // We do not process the `version` and `max_id` fields to see if the SystemReader processed
```

Comment:
This comment seems off; I don't see a `max_id` above, and it looks like it _does_ process `version`.